src/main/java/com/candyshop/islodycze/product/ProductService.java

```diff
@@ -0,0 +1,46 @@
+package com.candyshop.islodycze.product;
+
+import com.candyshop.islodycze.exceptions.ApplicationException;
+import com.candyshop.islodycze.model.Product;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@AllArgsConstructor
+@Service
+public class ProductService {
+
+    private final ProductRepository productRepository;
+
+    public Product createProduct(final Product product) {
+        if (productRepository.findByProductName(product.getProductName()) != null) {
+            throw new ApplicationException("Product with name " + product.getProductName() + " already exists.");
+        }
+        return productRepository.save(product);
+    }
+
+    public Product getProductById(Long productId) {
+        return productRepository.findById(productId)
+                .orElseThrow(() -> new ApplicationException(("Product with given ID: " + productId + " was not found.")));
+    }
+
+    public List<Product> getAllProducts() {
+        return productRepository.findAll();
+    }
+
+    public Product updateProduct(final Product product) {
+        if(productRepository.findById(product.getProductId()).isEmpty()) {
```

Comment:
Extract this train from `if` to a privet method. Mbe `isProductExist` or smth like that